pkg/build/scanner.go

```diff
@@ -88,6 +89,50 @@ func Scanner(manifest model.Manifest) error {
 		return nil
 	}
 
-	// There were vulnerabilities. Return body.
-	return errors.New("Image name: " + baseImageName + "\n" + string(body))
+	// There were vulnerabilities. Output message listing vulnerabilities.
+	log.Infof("Base image scan of %s failed with:\n %s", baseImageName, string(body))
+
+	// If IgnoreVulernability is true, just just return
+	if manifest.IgnoreVulnerability == true {
+		return nil
+	}
+
+	// Else build a new set of images.
+	// baseVersion is like 1.10-dev.1
+	// Increment the digit after the last period to get the new tag.
+	index := strings.LastIndex(baseVersion, ".") + 1
+	var digit int
+	if digit, err = strconv.Atoi(baseVersion[index:]); err != nil {
+		return err
+	}
+	newBaseVersion := baseVersion[:index] + strconv.Itoa(digit+1)
+	log.Infof("new baseVersion: %s", newBaseVersion)
+
+	// Attempted to run the ./tools/build-base-images.sh but locally, that gave me a
+	// number of different failures related to docker, PATHS, TTY, etc.
+	// Instead run the make command similar to what the script does.
+	buildEnv := []string{
+		"HUBS=docker.io/istio gcr.io/istio-release",
+		"TAG=" + newBaseVersion,
```

Comment:
there is a small chance this could overwrite an existing image. Is it worth adding some check? `docker manifest inspect` can do it but not sure if there are better options. `docker manifest inspect` is also slow so we shouldn't put it in `dockerx.pushx` - but maybe `./tools/build-base-images.sh
`

Reply:
I'll see if I can add it to the script.

Reply:
Thoughts on checking if all images might already exist, or can one simply check istio/base?

Reply:
probably just one is good enough? If its easier all of them is also fine, but it would be odd for that to happen

Reply:
https://github.com/istio/istio/pull/31323 was merged which iterates through and verifies we won't end up overwriting an image.